cairo/tests/utils/serde.py

```diff
@@ -427,28 +426,16 @@ def serialize_type(self, path: Tuple[str, ...], ptr) -> Any:
             return python_cls(value.to_bytes(python_cls.LENGTH, "little"))
 
         if python_cls == BNF12:
-            # BNF12 is represented as a struct with 12 UInt384
-            # We need to extract these values and create a BNF12 object
-            if value is None:
-                return None
+            # BNF12 is represented as a struct with 12 U384
 
-            coeffs = [
-                uint384_to_int(
-                    value[f"c{i}"]["d0"],
-                    value[f"c{i}"]["d1"],
-                    value[f"c{i}"]["d2"],
-                    value[f"c{i}"]["d3"],
-                )
-                for i in range(12)
-            ]
+            # In python, BNF12 is a tuple of 12 int but in cairo it's a struct with 12 U384
+            # Cast the U384 to int to be able to serialize
+            coeffs = [int(value[f"c{i}"]) for i in range(12)]
```

Comment:
isnt it just value.values()

Reply:
in which case this could simply be

```
        if python_cls == BNF12:
            value = value.values()
```

and then let the `signature(python_cls.__init__).bind(None, *value)` take over

Reply:
You need to cast to int as you receive a dict of U384 but BNF12 accept a tuple of int

Reply:
but it doesnt really matter